### PR TITLE
add unit test for ValidateStatefulSet

### PR DESCRIPTION
**Pull request by yuexiao-wang**
Signed-off-by: yuexiao-wang <wang.yuexiao@zte.com.cn>

**What this PR does / why we need it**:
add unit test for ValidateStatefulSetSpec

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by thockin** [comment]
Assign back to me once it is LGTM'ed and ready for approval.

**Comment by yuexiao-wang** [comment]
ping @kow3ns @tallclair

File: pkg/apis/apps/validation/validation_test.go [hunk 1]
```diff
@@ -818,3 +818,176 @@ func TestValidateControllerRevisionUpdate(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateStatefulSetSpec(t *testing.T) {
```
Comment (dhilipkumars) [inline pkg/apis/apps/validation/validation_test.go]: Hi! its not clear to me, how different is this from the `TestValidateStatefulSet` the first test case? could you elaborate on that please?

Comment (yuexiao-wang) [reply pkg/apis/apps/validation/validation_test.go]: I found there are some cases no covered in ValidateStatefulSetSpec. So I added TestValidateStatefulSetSpec to improve coverage.

File: pkg/apis/apps/validation/validation_test.go [hunk 2]
```diff
@@ -818,3 +818,176 @@ func TestValidateControllerRevisionUpdate(t *testing.T) {
...
+				Template:            validPodTemplate.Template,
+				UpdateStrategy: apps.StatefulSetUpdateStrategy{
+					Type:          apps.RollingUpdateStatefulSetStrategyType,
+					RollingUpdate: &apps.RollingUpdateStatefulSetStrategy{Partition: -1},
+				},
+			},
+		},
+	}
+
+	for _, errorCase := range errorCases {
```
[the remark this answers is not in the document]
Comment (yuexiao-wang) [reply pkg/apis/apps/validation/validation_test.go]: Got it. and I wil update

File: pkg/apis/apps/validation/validation_test.go (hunk 1 above)
Comment (dhilipkumars) [reply pkg/apis/apps/validation/validation_test.go]: Could these be added as a part of `TestValidateStatefulSet` to reduce code repetition? also as mentioned before if you could convert the test case to `go sub-test`,  testcases would be more informative.

**Comment by yuexiao-wang** [comment]
@dhilipkumars The PR is updated

**Comment by yuexiao-wang** [comment]
@dhilipkumars This PR is updated and PTAL

**Comment by dhilipkumars** [comment]
Please squash your commits
/lgtm from myside.

**Comment by yuexiao-wang** [comment]
@dhilipkumars It have done to squash these commits

**Comment by dhilipkumars** [comment]
cc @kubernetes/sig-apps-api-reviews PTAL and approve.

**Pull request merged**